plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/OpenShiftServerEditorSection.java

```diff
@@ -545,26 +543,35 @@ protected IStatus run(IProgressMonitor monitor) {
 								return Status.OK_STATUS;
 							}
 						})
-				.runWhenDone(
-						new Job("Setting connection, deploy project...") {
+				.runWhenDoneIf(job -> { 
+							if (connection == null) {
+								container.getDisplay().syncExec(() -> 
```

Comment:
Great. I have a couple of questions/suggestions that may be put to a new issue or ignored.

Will editor reload if user goes and creates the missing connection,or user has to reopen it manually? The other way, how the successfully loaded editor will react on deleting the connection? In this case it might be good not to show error dialog, but add an error section in the upper part of the editor. That section may be shown when dialog is shown too. Also, it may have active one-click solutions to the problem (e.g. buttons):
- Create the connection (wizard prefilled with host and username)
- Delete server adapter